model_analyzer/perf_analyzer/perf_analyzer.py

```diff
@@ -54,21 +55,35 @@ class PerfAnalyzer:
     with perf_analyzer.
     """
 
-    # The metrics that PerfAnalyzer can collect
-    perf_metrics = {
-        PerfLatencyAvg: "_parse_perf_latency_avg",
-        PerfLatencyP90: "_parse_perf_latency_p90",
-        PerfLatencyP95: "_parse_perf_latency_p95",
-        PerfLatencyP99: "_parse_perf_latency_p99",
-        PerfLatency: "_parse_perf_latency",
-        PerfThroughput: "_parse_perf_throughput",
-        PerfClientSendRecv: "_parse_perf_client_send_recv",
-        PerfClientResponseWait: "_parse_perf_client_response_wait",
-        PerfServerQueue: "_parse_perf_server_queue",
-        PerfServerComputeInfer: "_parse_perf_server_compute_infer",
-        PerfServerComputeInput: "_parse_perf_server_compute_input",
-        PerfServerComputeOutput: "_parse_perf_server_compute_output"
-    }
+    METRIC_TAG, CSV_STRING, RECORD_CLASS, REDUCTION_FACTOR = 0, 1, 2, 3
+    perf_metric_table = [
+        ["perf_latency_avg", "Avg latency", PerfLatencyAvg, 1000],
+        ["perf_latency_p90", "p90 latency", PerfLatencyP90, 1000],
+        ["perf_latency_p95", "p95 latency", PerfLatencyP95, 1000],
+        ["perf_latency_p99", "p99 latency", PerfLatencyP99, 1000],
+        ["perf_throughput", "Inferences/Second", PerfThroughput, 1],
+        ["perf_client_send_recv", "request/response", PerfClientSendRecv, 1000],
+        ["perf_client_send_recv", "send/recv", PerfClientSendRecv, 1000],
+        [
+            "perf_client_response_wait", "response wait",
+            PerfClientResponseWait, 1000
+        ], ["perf_server_queue", "Server Queue", PerfServerQueue, 1000],
+        [
+            "perf_server_compute_infer", "Server Compute Infer",
+            PerfServerComputeInfer, 1000
+        ],
+        [
+            "perf_server_compute_input", "Server Compute Input",
+            PerfServerComputeInput, 1000
+        ],
+        [
+            "perf_server_compute_output", "Server Compute Output",
+            PerfServerComputeOutput, 1000
+        ]
+    ]
+
+    # FIXME: This should be RECORD_CLASS, but I can't get the syntax to work
```

Comment:
resolve fixme

Reply:
2 possible workarounds, both from here: https://stackoverflow.com/questions/13905741/accessing-class-variables-from-a-list-comprehension-in-the-class-definition

1) Use a temp lambda function. This worked for me:
`
    perf_metrics = (lambda x=RECORD_CLASS,y=perf_metric_table: 
        [perf_metric[x] for perf_metric in y])()
`

2) make them member variables instead of static, and create them in __init__